components/brave_new_tab_ui/hooks/usePref.ts

```diff
@@ -0,0 +1,12 @@
+import { addChangeListener, getPreferences, sendSavePref } from '../api/preferences'
+import { PrefHookManager, usePref } from './PrefHookManager'
+
+const prefManager = new PrefHookManager(getPreferences, sendSavePref, addChangeListener)
+
+/**
+ * A hook for subscribing to and updating |NewTab.Preferences|.
+ * @param pref The name of the pref.
+ * @returns An array with two items, the current value of the pref, and a function
+ * for updating the pref.
+ */
+export const useNewTabPref = <T extends keyof NewTab.Preferences>(pref: T) => usePref(prefManager, pref)
```

Comment:
nit: `export function useNewTabPref<T...`

Reply:
I wonder if `T` / `keyof NewTab.Preferences` can be provided to the `PrefHookManager` and `usePref` infers `T` from the passed-in `PrefHookManager` instance? I know you're only providing it once here, but it seems like `usePref` _has_ to be given the correct `PrefHookManager` that matches the generic type passed to it, so it might as well just infer it from the `PrefHookManager`?

Reply:
Sure, I think with a minor refactor we can probably get rid of the extra type definitions, which should make things a bit cleaner. Basically this hook is currying `usePref` so I think if I expose a `createPrefHook` instead of `usePref` this should work nicely.

Good call! That'll be a nicer API :smile:

Reply:
Out of interest
```ts
function myFunctionOne(arg1, arg2) {
    // My clever algorithm
	return impressiveCalculation
}

// Would we always prefer this:
export function myWrapper(arg1) {
    return myFunctionOne(arg1, 7)
}

// Over this:
export const myWrapper = (arg1) => myFunctionOne(arg1, 7);

Reply:
Well, I suppose not for a 1-liner where it's simply calling another named function. Usually I prefer named functions primarily for stack traces, and secondarily for readability (something that get's called vs something assigned. And prefer arrow functions for one-off anonymous functions (e.g. map, filter) or where `this` needs to be preserved. It's personal opinion, and I don't think we've codified anything via the linter. So if I nitpick it and you don't agree then feel free to leave it.